src/dtx/dtx_internal.h

```diff
@@ -76,14 +77,18 @@ CRT_RPC_DECLARE(dtx, DAOS_ISEQ_DTX, DAOS_OSEQ_DTX);
  * dci_hints is sparse array, one per engine, sorted against the rank ID.
  * It can hold more than 19K engines inline RPC body.
  */
+/* clang-format off */
 #define DAOS_ISEQ_COLL_DTX						\
 	((uuid_t)		(dci_po_uuid)		CRT_VAR)	\
 	((uuid_t)		(dci_co_uuid)		CRT_VAR)	\
 	((struct dtx_id)	(dci_xid)		CRT_VAR)	\
 	((uint32_t)		(dci_version)		CRT_VAR)	\
+	((uint32_t)		(dci_min_rank)		CRT_VAR)	\
+	((uint32_t)		(dci_max_rank)		CRT_VAR)	\
```

Comment:
will this cause interoperability issue with different version of client / server?

Reply:
This is RPC among servers, will not cause interoperability issue between client and server.

> the PR should renable coll RPC by default for punch and stat too?
> otherwise we also don't know since it's not tested in CI or other places

Collective punch is disabled only on release/2.6, not on master. The patch for 2.6 (https://github.com/daos-stack/daos/pull/15975) enables collective punch explicitly.

Reply:
i see. thanks for explaining.